src/dmd/init.d

```diff
@@ -36,15 +36,30 @@ enum NeedInterpret : int
 alias INITnointerpret = NeedInterpret.INITnointerpret;
 alias INITinterpret = NeedInterpret.INITinterpret;
 
+/*************
+ * Discriminant for which kind of initializer
+ */
+enum InitKind : ubyte
```

Comment:
Would this information be useful to C++ consumers? Or should they only be using visitors/the available isXxxInitializer() methods to get to the correct type?

Reply:
I don't know. I'm not even sure why a C++ consumer would need init.h at all.

Reply:
To generate code for the _init of a variable.

Having a quick/naive look (I have probably missed something), I can only see three points of interest.
- isVoidInitializer
- isExpInitializer
- initializerToExpression

Reply:
If your code sticks with the `is` functions, I don't think you'll need the enum at all.